supreme/src/androidMain/kotlin/at/asitplus/signum/supreme/os/AndroidKeyStoreProvider.kt

```diff
@@ -366,34 +322,49 @@ sealed class LockedAndroidKeystoreSigner private constructor(
         }
     }
 
-    protected abstract fun toUnlocked(jcaSig: JCASignatureObject): UnlockedAndroidKeystoreSigner
-
-    @HazardousMaterials
-    final override suspend fun unlock(): KmmResult<UnlockedAndroidKeystoreSigner> =
-        signatureAlgorithm.getJCASignatureInstance().onSuccess {
+    internal suspend fun getJCASignature(signingConfig: AndroidSignerSigningConfiguration): Signature =
+        signatureAlgorithm.getJCASignatureInstance().getOrThrow().also {
             if (needsAuthenticationForEveryUse) {
                 it.initSign(jcaPrivateKey)
-                attemptBiometry(config.unlockPrompt.v, CryptoObject(it))
+                attemptBiometry(DSL.ConfigStack(signingConfig.unlockPrompt.v, config.unlockPrompt.v), CryptoObject(it))
             } else {
                 try {
                     it.initSign(jcaPrivateKey)
                 } catch (_: UserNotAuthenticatedException) {
-                    attemptBiometry(config.unlockPrompt.v, null)
+                    attemptBiometry(DSL.ConfigStack(signingConfig.unlockPrompt.v, config.unlockPrompt.v), null)
                     it.initSign(jcaPrivateKey)
                 }
             }
-        }.mapCatching(this::toUnlocked)
+        }
+
+    final override suspend fun trySetupUninterruptedSigning(configure: DSLConfigureFn<AndroidSignerSigningConfiguration>) = catching {
+        if (needsAuthentication && !needsAuthenticationForEveryUse) {
+            getJCASignature(DSL.resolve(::AndroidSignerSigningConfiguration, configure))
+        }
+    }
+
+    final override suspend fun sign(
+        data: SignatureInput,
+        configure: DSLConfigureFn<AndroidSignerSigningConfiguration>
+    ): KmmResult<CryptoSignature> = catching {
```

Comment:
I maintain that this should return a custom ternary sealed result class:
* successful signature CryptoSignature is part of the result
* auth failed -> maybe with a string explanation)
* error -> contains a throwable

The same goes for iOS. On the JVM the same could happen for some non-standard procedures, so it does make sense that also the JVM returns such a ternary result.